tar/lib/src/reader.dart

```diff
@@ -0,0 +1,465 @@
+// Copyright 2020
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+
+import 'constants.dart';
+import 'exceptions.dart';
+import 'format.dart';
+import 'header.dart';
+import 'header_impl.dart';
+import 'sparse_entry.dart';
+import 'stream.dart';
+import 'utils.dart';
+
+/// [TarReader] provides sequential access to the TAR files in a TAR archive.
+/// It is designed to read from a stream and to spit out substreams for
+/// individual file contents in order to minimize the amount of memory needed
+/// to read each archive where possible.
+@sealed
+class TarReader {
+  /// A chunked stream iterator to enable us to get our data.
+  final ChunkedStreamIterator<int> _chunkedStream;
+
+  /// The [TarHeader] for the current file. Completes when a call to `next`
+  /// completes as well.
+  ///
+  /// Is `null` before the first call to `next` and after a call to `next`
+  /// completes with a `false` result.
```

Comment:
False result?

Reply:
haha - completes with `false`. I tried updating the documentation, let me know if the new one works better or still suffers from the same issue.